src/claimtrie.cpp

```diff
@@ -41,6 +41,68 @@ uint256 getValueHash(const COutPoint& outPoint, int nHeightOfLastTakeover)
     return result;
 }
 
+template <typename T>
+bool equals(const T& value, const COutPoint& outPoint)
+{
+    return value.outPoint == outPoint;
+}
+
+template <typename T>
```

Comment:
Can we make CSupportValue and CClaimValue have the same base class that has the name and outpoint so that we don't have to use templated methods to work with them?

Reply:
We can but dynamic dispatch is runtime / slower vs static one. If we go for polymorphic approach we'll face same penalty as well as runtime check to determine actual type of parameter.

Reply:
Taking a base class as a parameter doesn't automatically make a virtual function, no?

Reply:
Yes, there is not automatically virtual functions, but you get the base class at some point of implementation you should what exactly is this type, claim or support without virtual dispatch you can't do it.

Reply:
The C approach is same type for both (claim and support) with field in (enum in most cases) that describe the actual type.